bug(oxc-transform): Isolated Declarations Incorrectly Preserves Default Values in Rest Parameters

## Description

When using `oxc-transform`'s isolated declarations feature (`typescript.declaration: true`), default values in rest parameters are incorrectly preserved in the generated `.d.ts` files. This causes TypeScript compiler errors when these declaration files are consumed, as parameter initializers are not allowed in declaration files.

## Environment

- **oxc-transform version**: Latest (as of 2026-01-02)
- **TypeScript version**: 5.x
- **tsconfig.json**: `isolatedDeclarations: true`

## Reproduction

### Source Code (`dateTime.ts`)

```typescript
export function formatDate(
  config: {
    locale: string;
    timeZone?: string;
  },
  getDateTimeFormat: SomeType,
  ...[value, options = {}]: Parameters<SomeFormatter['formatDate']>
): string {
  // implementation
}
```

### oxc-transform Configuration

```javascript
import { transformSync } from 'oxc-transform';

const result = transformSync(filePath, code, {
  lang: 'ts',
  sourcemap: false,
  typescript: {
    onlyRemoveTypeImports: true,
    declaration: true,
  },
});

// result.declaration contains the generated .d.ts
```

### Generated Declaration File (Incorrect)

```typescript
export declare function formatDate(
  config: {
    locale: string;
    timeZone?: string;
  },
  getDateTimeFormat: SomeType,
  ...[value, options = {}]: Parameters<SomeFormatter["formatDate"]>
): string;
```

### Expected Declaration File (Correct)

```typescript
export declare function formatDate(
  config: {
    locale: string;
    timeZone?: string;
  },
  getDateTimeFormat: SomeType,
  ...args: Parameters<SomeFormatter["formatDate"]>
): string;
```

Or better yet, the rest parameter should be expanded:

```typescript
export declare function formatDate(
  config: {
    locale: string;
    timeZone?: string;
  },
  getDateTimeFormat: SomeType,
  value: Parameters<SomeFormatter["formatDate"]>[0],
  options?: Parameters<SomeFormatter["formatDate"]>[1]
): string;
```

## Error

When consuming the generated `.d.ts` file, TypeScript reports:

```
error TS2371: A parameter initializer is only allowed in a function or constructor implementation.
```

## Impact

This bug prevents using oxc-transform's isolated declarations feature as a drop-in replacement for `tsc --emitDeclarationOnly` when the source code contains rest parameters with default values.

## Workaround

The only workaround is to modify the source code to avoid rest parameter syntax with defaults:

```typescript
// Instead of:
...[value, options = {}]: Parameters<SomeFormatter['formatDate']>

// Use explicit parameters:
value: Parameters<SomeFormatter['formatDate']>[0],
options: Parameters<SomeFormatter['formatDate']>[1] = {}
```

## Context

This issue was discovered while migrating a large monorepo ([formatjs](https://github.com/formatjs/formatjs)) to use oxc-transform for both JavaScript transpilation and declaration generation. The rest parameter pattern with defaults is relatively common in TypeScript codebases.

## Related

- TypeScript's `isolatedDeclarations` mode should not preserve implementation details like default values in declaration files
- The rest parameter syntax is a convenient way to maintain type safety with variadic functions
- This affects any TypeScript project using isolated declarations with oxc-transform

## Expected Behavior

oxc-transform's isolated declarations should either:
1. Strip default values from rest parameters in generated `.d.ts` files, OR
2. Expand rest parameters into individual parameters without default values in the declaration

---

**Sources:**
- [oxc-transform npm package](https://www.npmjs.com/package/oxc-transform)
- [Oxc Transformer Alpha announcement](https://oxc.rs/blog/2024-09-29-transformer-alpha)
- [TypeScript Transformer docs](https://oxc.rs/docs/guide/usage/transformer/typescript)

--- a/crates/oxc_isolated_declarations/src/function.rs
+++ b/crates/oxc_isolated_declarations/src/function.rs
@@ -160,12 +160,16 @@ impl<'a> IsolatedDeclarations<'a> {
             self.error(parameter_must_have_explicit_type(rest.span));
         }
 
-        self.ast.alloc_formal_parameters(
-            params.span,
-            FormalParameterKind::Signature,
-            items,
-            params.rest.clone_in(self.ast.allocator),
-        )
+        let rest = params.rest.as_ref().map(|rest| {
+            let mut rest = rest.clone_in(self.ast.allocator);
+            FormalParameterBindingPattern::remove_assignments_from_kind(
+                self.ast,
+                &mut rest.rest.argument,
+            );
+            rest
+        });
+
+        self.ast.alloc_formal_parameters(params.span, FormalParameterKind::Signature, items, rest)
     }
 }
 

--- a/crates/oxc_isolated_declarations/tests/fixtures/function-parameters.ts
+++ b/crates/oxc_isolated_declarations/tests/fixtures/function-parameters.ts
@@ -32,3 +32,6 @@ export function withAny(a: any = 1, b: string): void { }
 export function withUnknown(a: unknown = 1, b: string): void { }
 
 export function withTypeAssertion(a = /regular-repression-cannot-infer/ as any): void { }
+
+export function restWithDefault(...[value, options = {}]: [string, object]): void { }
+export function restWithNestedDefault(...[a, { b = 1, c = 2 }]: [number, { b: number, c: number }]): void { }

